### PR TITLE
Creating the Settings Layout Page and Settings Navigation tests

### LINKED ISSUE #12: Create the layout for the settings page
Create the layout for the settings page according to the screenshots. You don't need to implement any functionalities.

- [x] Tracking preferences section has four icon buttons with texts given by the screenshot
- [x] Notifications and Account Settings section has a list of tabs.
- [x] Clicking on each of the tab should take you to an empty page
- [x] The page is scrollable
Screen Shot 2022-11-06 at 2 31 19 PM

### PR DESCRIPTION
**Pull request by manya-mittal**
In this pull request: 

- Made the Settings page's layout 
- Clicking on the tabs (like the "Customise Notifications") will lead to empty pages 
- The toggle button's state is saved 
- The Toggle buttons turn green if you click on them and go back to white if you click again 
- Made some tests for the navigation

Once the other pull requests are merged, I can move some of the code around like the navigation host to the AppScreen. I think I would need the bottom navigation bar to be there first though.

### CODE REVIEW

**Comment by michaelng0107** [comment]
@manya-mittal ~~could you merge develop into this so we can see what it looks like after it merges? Let us know if you need any help resolving conflicts~~
Nevermind, I forgot to pull lol

File: app/src/main/java/com/example/theperiodpurse/AppScreen.kt [hunk 1]
```diff
@@ -44,6 +50,13 @@ fun ScreenApp(
             }
         }
     ) { innerPadding ->
+        Image(
+            painter = painterResource(id = R.drawable.background),
+            contentDescription = null,
+            modifier = Modifier.fillMaxWidth().fillMaxHeight().fillMaxSize(),
```
Comment (michaelng0107) [inline app/src/main/java/com/example/theperiodpurse/AppScreen.kt]: nitpick: I think `fillMaxSize()` already does `fillMaxWidth` and `fillMaxHeight`, so having the other two would be redundant.

File: app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    onDeleteClicked: () -> Unit,
+){
+   Column(modifier = modifier
+       .fillMaxSize()
+       .verticalScroll(rememberScrollState())) {
+       Text(
+           text = stringResource(R.string.tracking_preferences),
+           modifier = modifier.padding(
+               top = 50.dp,
+               start = 10.dp),
```
Comment (michaelng0107) [inline app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt]: if we are using `top = 50.dp` and `start = 10.dp` in every Text composable, would it be a good idea if we defined something like `val textModifier = modifier.padding(top = 50.dp, start = 10.dp)` and reuse it in the Text composables?

File: app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+           text = stringResource(R.string.back_up_account),
+           onTabClicked = onBackUpClicked
+       )
+       Spacer(modifier = Modifier.padding(10.dp))
+       TabOption(
+           text = stringResource(id = R.string.delete_account),
+           onTabClicked = onDeleteClicked
+       )
+       Spacer(modifier = Modifier.padding(10.dp))
+       Text(text = "The Period Purse. All rights reserved.")
```
Comment (michaelng0107) [inline app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt]: perhaps use a string resource here instead? Looks like something we will reuse in other places

File: app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        horizontalArrangement = Arrangement.Center) {
+        TrackingOptionButton(
+            modifier = modifier,
+            label = stringResource(R.string.mood),
+            icon = painterResource(id = R.drawable.sentiment_neutral_black_24dp)
+        )
+        TrackingOptionButton(
+            modifier = modifier,
+            label = stringResource(R.string.exercise),
+            icon = painterResource(id = R.drawable.self_improvement_black_24dp)
```
Comment (michaelng0107) [inline app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt]: Looks like we will be using these label and icon pairs on a few different pages (`R.string.mood matches` with `R.drawable.sentiment_neutral_black_24dp`), such as the onboarding pages and the calendar page.

We can look into modifying the `symptoms` object in `DataSource` to include this information for reusability. Perhaps we can create an issue for this sort of refactoring/enhancement?

File: app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,) {
+        IconToggleButton(
+            checked = checked,
+            onCheckedChange = { checked = !checked},
+            modifier = Modifier.clip(RoundedCornerShape(20.dp))
+        ) {
+            Icon(
+                painter = icon,
+                contentDescription = "contentDescription",
```
Comment (michaelng0107) [inline app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt]: do we want "contentDescription" to be the content description of the button? Perhaps we can have something more informative like "click this to track/stop tracking symptom <label>" depending on whether this button is checked

File: app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenNavigation.kt [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+                currentScreen = currentScreen,
+                canNavigateBack = navController.previousBackStackEntry != null,
+                navigateUp = { navController.navigateUp() }
+            )
+        }
+    ) { innerPadding ->
+        Image(
+            painter = painterResource(id = R.drawable.background),
+            contentDescription = null,
+            modifier = Modifier.fillMaxWidth().fillMaxHeight().fillMaxSize(),
```
Comment (michaelng0107) [inline app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenNavigation.kt]: same as above about `fillMaxSize`

Comment (manya-mittal) [reply app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenNavigation.kt]: Thanks for the comments Michael, I've made the changes you mentioned.

File: app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt (hunk 4 above)
Comment (manya-mittal) [reply app/src/main/java/com/example/theperiodpurse/ui/setting/SettingScreenLayout.kt]: This is the only change that I didn't make yet. It seems like this should be for a different PR, but lmk if theres anything to change now as well

**Review by elenawzy — APPROVED** [review]
approved!!

**Review by leowrites — APPROVED** [review]

**Pull request merged**